crates/sui-framework/sources/publisher.move

```diff
@@ -103,6 +111,10 @@ module sui::test_publisher {
         assert!(publisher::is_package<CustomType>(&pub), 0);
         assert!(publisher::is_package<Scenario>(&pub), 0);
 
+        assert!(&ascii::string(b"0000000000000000000000000000000000000002") == publisher::package(&pub), 0);
```

Comment:
nit: These tests probably belong in the stdlib rather than here, as that's where most of the logic is -- but very happy to see them!

Reply:
That's a tricky part. We don't have tests in the stdlib and I never dared to add one. Guess it's a bigger question on whether it's okay to have tests for stdlib or there's a better place for them where they live?

Reply:
Yeah, interesting -- maybe @sblackshear / @tnowacki has context on why there seems to not be `#[test]`s in the stdlib?

Reply:
(Any case, not a blocker for this PR)